foundation/core/variables.tf

```diff
@@ -6,6 +6,11 @@ variable "location" {
   type = string
 }
 
+variable "env" {
+  type    = string
+  default = ""
```

Comment:
It kinda seems like we're treating this as a required variable, even though it's optional.  If users omit this, the names of all of the resources will have a double dash `--`, which I assume isn't what you intended.  Maybe we should make this required.